src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -709,7 +709,11 @@ public override bool Equals(object other)
         public override int GetHashCode()
 #pragma warning restore CS0809
         {
-            throw ExceptionUtilities.Unreachable;
+            if (_defaultType is null)
+            {
+                return 0;
+            }
+            return Hash.Combine(_defaultType.GetHashCode(), IsAnnotated.GetHashCode());
```

Comment:
Should we also include `_extensions` in the mix?

Reply:
Changed to use `TypeSymbol` and `IsAnnotated` to ensure `GetHashCode()` returns the same value for two instances that would compare `Equals()`.

---
In reply to: [207666724](https://github.com/dotnet/roslyn/pull/29045#discussion_r207666724) [](ancestors = 207666724)